### PR TITLE
Add proxies to the master and worker templates

### PR DESCRIPTION
**Pull request by tfussell**
## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/template/master_template.go [hunk 1]
```diff
@@ -138,6 +138,15 @@ systemd:
...
           Environment="DOCKER_CGROUPS=--exec-opt native.cgroupdriver=cgroupfs --cgroup-parent=/kubereserved.slice --log-opt max-size=25m --log-opt max-file=2 --log-opt labels=io.kubernetes.container.hash,io.kubernetes.container.name,io.kubernetes.pod.name,io.kubernetes.pod.namespace,io.kubernetes.pod.uid"
           Environment="DOCKER_OPT_BIP=--bip={{.Cluster.Docker.Daemon.CIDR}}"
+          {{- if .Proxy.HTTP }}
+          Environment="HTTP_PROXY={{ .Proxy.HTTP }}"
+          {{- end }}
+          {{- if .Proxy.HTTPS }}
+          Environment="HTTPS_PROXY={{ .Proxy.HTTPS }}"
+          {{- end }}
+          {{- if .Proxy.NoProxy }}
+          Environment="NO_PROXY={{ .Proxy.NoProxy }}"
```
Comment (glitchcrab) [inline pkg/template/master_template.go:148]: so we're passing the values as a single comma-delimited string? i assume this string will be created from the installation config by kvm-operator?

Comment (yulianedyalkova) [reply pkg/template/master_template.go:148]: > so we're passing the values as a single comma-delimited string?

Yes

> i assume this string will be created from the installation config by kvm-operator?

`kvm-operator` will pass the values from its environment ([link](https://github.com/giantswarm/kvm-operator/blob/be5d19eddf8cae39b11976c35b79185d6d645e6d/service/controller/cloudconfig/worker_template.go#L45-L47)). So flow is `installations` -> `kyverno` -> `kvm-operator` -> `k8scloudconfig`.

Comment (tfussell) [reply pkg/template/master_template.go:148]: I thought about making it a list and joining it as we do elsewhere, but it's already a comma-delimited string so it wouldn't gain us much. We'll pass it in via config in a nicer way in the future: https://github.com/giantswarm/giantswarm/issues/17252

**Comment by tfussell** [comment]
I can't approve as I opened the PR for Yulia, but we've tested it and it looks good to me.

**Review by glitchcrab — APPROVED** [review]
Lgtm

**Pull request merged**